packages/block-library/src/gallery/gallery-image.js

```diff
@@ -127,6 +127,23 @@ class GalleryImage extends Component {
 		/* eslint-disable jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/onclick-has-role, jsx-a11y/click-events-have-key-events */
 		return (
 			<figure className={ className } tabIndex="-1" onKeyDown={ this.onKeyDown } ref={ this.bindContainer }>
+				{ process.env.GUTENBERG_PHASE === 2 ?
+					isSelected && <div className="block-library-gallery-item__move-menu">
+						<IconButton
+							icon="arrow-left"
+							onClick={ onMoveBackward }
+							className="blocks-gallery-item__move-backward"
+							label={ __( 'Move Image Backward' ) }
+						/>
+						<IconButton
+							icon="arrow-right"
+							onClick={ onMoveForward }
+							className="blocks-gallery-item__move-forward"
+							label={ __( 'Move Image Forward' ) }
```

Comment:
Would it be possible to disable the button if it cannot go more to that side?

Reply:
yeah!

Reply:
ed24ea2